spark_log_parser/parsing_models/application_model.py

```diff
@@ -194,20 +195,25 @@ def __init__(self, eventlogpath, bucket=None, stdoutpath=None, debug=False):  #
 
                 elif event_type == "SparkListenerEnvironmentUpdate":
 
-                    curKeys = json_data["Spark Properties"].keys()
+                    spark_properties = json_data["Spark Properties"]
+                    event_keys = spark_properties.keys()
 
                     # This if is specifically for databricks logs
-                    if "spark.databricks.clusterUsageTags.sparkVersion" in curKeys:
-
+                    if "spark.databricks.clusterUsageTags.sparkVersion" in event_keys:
                         self.cloud_platform = "databricks"
-                        self.cloud_provider = json_data["Spark Properties"][
+                        self.cluster_id = spark_properties["spark.databricks.clusterUsageTags.clusterId"]
+                        self.cloud_provider = spark_properties[
                             "spark.databricks.clusterUsageTags.cloudProvider"
                         ].lower()
-                        self.spark_version = json_data["Spark Properties"][
+                        self.spark_version = spark_properties[
                             "spark.databricks.clusterUsageTags.sparkVersion"
                         ]
+                    else:
+                        self.cloud_platform = "emr"
+                        self.cloud_provider = "aws"
+                        self.cluster_id = json_data["System Properties"]["EMR_CLUSTER_ID"]
```

Comment:
For EMR logs, this cluster ID is under the `System Properties` object of the event.